src/Columns/IColumn.h

```diff
@@ -462,10 +463,10 @@ class IColumn : public COW<IColumn>
 
     /// Returns column with @total_size elements.
     /// In result column values from current column are at positions from @offsets.
-    /// Other values are filled by @default_value.
+    /// Other values are filled by value from @column_with_default_value.
     /// @shift means how much rows to skip from the beginning of current column.
     /// Used to create full column from sparse.
-    [[nodiscard]] virtual Ptr createWithOffsets(const Offsets & offsets, const Field & default_field, size_t total_rows, size_t shift) const;
+    [[nodiscard]] virtual Ptr createWithOffsets(const Offsets & offsets, const ColumnConst & column_with_default_value, size_t total_rows, size_t shift) const;
```

Comment:
I changed `Field` to `ColumnConst` because in `Variant` column we cannot properly work with `Field` (it's not trivial to determine in which variant we need to insert the value). Working with `ColumnConst` with single value solves this problem.